### ISSUE TITLE
Consider returning the original revert message with TransferHelper

### ISSUE THREAD

**Issue by albertocuestacanada** [Uniswap/uniswap-v3-periphery#50]
TransferHelper overwrites the revert message, which makes it impossible to know whether the transaction failed because of insufficient balance, insufficient approval, or something else.

The code to return the revert message is [here](https://github.com/Uniswap/uniswap-v3-periphery/blob/3ea441b548528e697e30415966b81e24f26add2a/contracts/base/Multicall.sol#L16).

This issue applies as well to TransferHelper in the [core repo](https://github.com/Uniswap/uniswap-v3-core/blob/main/contracts/libraries/TransferHelper.sol).

**Comment by moodysalem** [comment]
True, but also no transaction should be constructed and submitted to the chain s.t. the allowance is insufficient or the balance is insufficient, so the revert reason isn't particularly helpful for users.

I'd consider this low priority, definitely won't change in core

**Comment by albertocuestacanada** [comment]
Yeah, with the testing and the audits done it's probably too late to fix this. Maybe it will come useful in v4.
